Add support for list of classes in `DT2BSClass`

pedropark99 opened the pull request:
This PR seeks to bring support for a list of classes in `DT2BSClass()`, with the end goal of fixing the issue #1052.

In more details, when you select `style='bootstrap'`, `DT::datatable()` tries to translate the `class` argument, to a css rule with all the Bootstrap classes that were mentioned in the `class` argument. To make this translation, the `DT2BSClass()` function is called with the class argument. However, the `DT2BSClass()` function only supports a character vector as input.

This PR will allow `DT2BSClass()` to receive a list of booleans as input, as mentioned in the example from issue #1052.
```r
# Now these examples below works
DT::datatable(mtcars, class=list(stripe=F), style = "bootstrap")
DT::datatable(mtcars, class=list(stripe=T, compact=F, hover=T), style = "bootstrap")
```

In essence, we use `base::Filter()` to select the elements of the list that are TRUE, and simply extract the name of these filtered elements.
```r
  if (is.list(class)) {
    chosen_classes = Filter(function(x) isTRUE(x), class)
    class = names(chosen_classes)
  }
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rstudio/DT?pullRequest=1089) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rstudio/DT?pullRequest=1089) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rstudio/DT?pullRequest=1089) it.</sub>

yihui commented on R/datatables.R:
```diff
@@ -932,9 +932,17 @@ DTDependencies = function(style) {
-  class = unlist(strsplit(class, '\\s+'))
-  if ('display' %in% class)
+  if (is.list(class)) {
+    chosen_classes = Filter(function(x) isTRUE(x), class)
+    class = names(chosen_classes)
```
Perhaps this suffices?

```suggestion
    class = names(which(unlist(class)))
```

yihui reviewed (approved):
Looks good. It will be great to add a NEWS item to `NEWS.md`. Thanks!

pedropark99 commented:
Thank you @yihui  for the review! I updated NEWS.md with the changes, and, also changed the source code to use `which()` instead of `Filter()` as you suggested 😉
